src/async_op.c

```diff
@@ -35,30 +41,69 @@ static void async_op_dispose(ASYNC_OP* async_op)
 #define IS_POWER_OF_2(x) \
     (((x) & ((x) - 1)) == 0)
 
+/*tiny function that returns the next multiple of "align" starting from "start" (included). Does 1 mod and 1 branch*/
+static void* get_next_aligned_address(void* start, uint32_t align)
+{
+    void* result;
+    uint32_t m = (uintptr_t)start % align;
+    if (m == 0)
+    {
+        result = start;
+    }
+    else
+    {
+        result = (void*)((uintptr_t)start - m + align);
+    }
+    return result;
+}
+
+
+static void* get_prev_aligned_with_size_address(void* start, uint32_t size, uint32_t align)
+{

```

Comment:
No `void * result` like the above `get_next_aligned_address`?

Reply:
added void* result